sdk/core/Microsoft.Extensions.Azure/src/Internal/FallbackAzureClientFactory.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Azure.Core;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Extensions.Azure
+{
+    internal class FallbackAzureClientFactory<TClient>: IAzureClientFactory<TClient>
+    {
+        private readonly IOptionsMonitor<AzureClientsGlobalOptions> _globalOptions;
+        private readonly IServiceProvider _serviceProvider;
+        private readonly EventSourceLogForwarder _logForwarder;
```

Comment:
This seems to be unused, do we need to do something with it?

Reply:
Resolving the type initiates the even source subscription. I'll change it to be a little more explicit and easier to understand.

Reply:
Gotcha, thanks!